packages/admin/dashboard/src/components/data-grid/components/data-grid-text-cell.tsx

```diff
@@ -35,37 +47,95 @@ export const DataGridTextCell = <TData, TValue = any>({
 const Inner = ({
   field,
   inputProps,
+  isMultiLine,
 }: {
   field: ControllerRenderProps<any, string>
   inputProps: InputProps
+  isMultiLine: boolean
 }) => {
   const { onChange: _, onBlur, ref, value, ...rest } = field
   const { ref: inputRef, onBlur: onInputBlur, onChange, ...input } = inputProps
 
   const [localValue, setLocalValue] = useState(value)
+  const textareaRef = useRef<HTMLTextAreaElement>(null)
 
   useEffect(() => {
     setLocalValue(value)
   }, [value])
 
   const combinedRefs = useCombinedRefs(inputRef, ref)
+  const textareaCombinedRefs = useCombinedRefs(inputRef, ref, textareaRef)
+
+  const adjustTextareaHeight = useCallback(() => {
+    const textarea = textareaRef.current
+    if (textarea) {
+      // Reset height to 0 to get accurate scrollHeight
+      textarea.style.height = "0px"
+      // Set the height to match content (minimum 24px for min visible height)
+      const newHeight = Math.max(textarea.scrollHeight, 24)
+      textarea.style.height = `${newHeight}px`
+    }
+  }, [])
+
+  // Adjust height when value changes
+  useEffect(() => {
+    if (isMultiLine) {
+      adjustTextareaHeight()
+    }
+  }, [localValue, isMultiLine, adjustTextareaHeight])
+
+  useEffect(() => {
+    if (isMultiLine) {
+      // Immediate adjustment
+      adjustTextareaHeight()
+      // Delayed adjustment to handle any layout shifts
+      const timeoutId = setTimeout(adjustTextareaHeight, 50)
+      return () => clearTimeout(timeoutId)
+    }
+  }, [isMultiLine, adjustTextareaHeight])
+
+  if (isMultiLine) {
```

Comment:
Q: now that we decided to always show the whole content, in the long term will we still need the split between multi vs not multi?